demisto_sdk/commands/find_dependencies/find_dependencies.py

```diff
@@ -203,10 +204,45 @@ def _search_for_pack_items(pack_id: str, items_list: list) -> list:
         """
         return list(filter(lambda s: next(iter(s.values())).get('pack') == pack_id, items_list))
 
+    @staticmethod
+    def _is_item_match(item_details: dict, base_condition: bool, exclude_ignored_dependencies: bool, marketplace: str) -> bool:
+        """
+        Whether the item matches the criterias:
+            * matches the base condition.
+            * relevant to the marketplaces server versions.
+            * not in an excluded pack.
+            * part of the desired marketplce.
+
+        Args:
+            item_details: item info from the ID set.
+            base_condition: basic check of relevance, for example item ID is matched.
+            exclude_ignored_dependencies: Determines whether to include unsupported dependencies or not.
+            marketplace: The dependency calculation desired marketplace.
+
+        Returns:
+            bool
+        """
+        id_set_item_version = Version(item_details.get('toversion', DEFAULT_CONTENT_ITEM_TO_VERSION))
+        is_version_match = id_set_item_version >= MINIMUM_DEPENDENCY_VERSION
+
+        id_set_item_pack = item_details.get('pack')
+        is_relevant_pack = bool(id_set_item_pack and
+                                (
+                                    not exclude_ignored_dependencies or
+                                    id_set_item_pack not in constants.IGNORED_DEPENDENCY_CALCULATION
+                                ))
+
+        is_marketplace_match = not marketplace or marketplace in item_details.get('marketplaces', [])
+
+        return base_condition and is_version_match and is_relevant_pack and is_marketplace_match
+
     @staticmethod
     def _search_packs_by_items_names(items_names: Union[str, list],
```

Comment:
This method (and _is_item_match) are called many times.
Maybe their response can be cached or something?